nodejs/eks/cni-addon.ts

```diff
@@ -0,0 +1,545 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as k8s from "@pulumi/kubernetes";
+import * as deepmerge from "deepmerge";
+
+export interface VpcCniAddonOptions {
+    clusterName: pulumi.Input<string>;
+
+    /**
+     * The Kubernetes version of the cluster. This is used to determine the addon version to use if `addonVersion` is not specified.
+     */
+    clusterVersion?: pulumi.Input<string>;
+
+    /**
+     * How to resolve field value conflicts when migrating a self-managed add-on to an Amazon EKS add-on. Valid values are `NONE` and `OVERWRITE`. For more details see the [CreateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateAddon.html) API Docs.
+     */
+    resolveConflictsOnCreate?: pulumi.Input<string>;
+
+    /**
+     * How to resolve field value conflicts for an Amazon EKS add-on if you've changed a value from the Amazon EKS default value. Valid values are `NONE`, `OVERWRITE`, and `PRESERVE`. For more details see the [UpdateAddon](https://docs.aws.amazon.com/eks/latest/APIReference/API_UpdateAddon.html) API Docs.
+     */
+    resolveConflictsOnUpdate?: pulumi.Input<string>;
+
+    /**
+     * The Amazon Resource Name (ARN) of an
+     * existing IAM role to bind to the add-on's service account. The role must be
+     * assigned the IAM permissions required by the add-on. If you don't specify
+     * an existing IAM role, then the add-on uses the permissions assigned to the node
+     * IAM role. For more information, see [Amazon EKS node IAM role](https://docs.aws.amazon.com/eks/latest/userguide/create-node-role.html)
+     * in the Amazon EKS User Guide.
+     *
+     * > **Note:** To specify an existing IAM role, you must have an IAM OpenID Connect (OIDC)
+     * provider created for your cluster. For more information, [see Enabling IAM roles
+     * for service accounts on your cluster](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
+     * in the Amazon EKS User Guide.
+     */
+    serviceAccountRoleArn?: pulumi.Input<string>;
+
+    /**
+     * Key-value map of resource tags. If configured with a provider `defaultTags` configuration block present, tags with matching keys will overwrite those defined at the provider-level.
+     */
+    tags?: pulumi.Input<{
+        [key: string]: pulumi.Input<string>;
+    }>;
+
+    /**
+     * Custom configuration values for the vpc-cni addon. This object must match the schema derived from [describe-addon-configuration](https://docs.aws.amazon.com/cli/latest/reference/eks/describe-addon-configuration.html).
+     */
+    configurationValues?: pulumi.Input<object>;
```

Comment:
AWS offers schemas (jsonschema) for the addon configuration. But those can differ per addon version (and can include breaking changes).

I'm thinking that we could generate the types for this, but that requires a versioning strategy (forwards & backwards compatibility) and automation. The generation itself  should be doable, we're already doing that in aws-native.

I'd propose that we punt this for a later time and gauge user interest in the meantime.

Reply:
And this would be better placed in aws classic IMHO. That way the whole user base profits from it and not only pu-eks users

Reply:
We had a chat about this, opening an issue in aws classic for it

Reply:
Cool idea but I'd defer in an issue to get this off the critical path.